mep/common/views.py

```diff
@@ -43,6 +47,56 @@ def dispatch(self, request, *args, **kwargs):
         return response
 
 
+class RdfViewMixin(ContextMixin):
+    '''View mixin to add an RDF linked data graph to context for use in serializing
+    and embedding structured data in templates.'''
+
+    #: default schema.org type for a View
+    rdf_type = SCHEMA_ORG.WebPage
+    #: breadcrumbs, used to render breadcrumb navigation. they should be a list
+    #: of tuples like ('Title', '/url')
+    breadcrumbs = []
+    #: json-ld context for the rdf graph; defaults to schema.org
+    json_ld_context = str(SCHEMA_ORG)
+
+    def get_context_data(self, *args, **kwargs):
+        '''Add generated breadcrumbs and an RDF graph to the view context.'''
+        context = super().get_context_data(*args, **kwargs)
+        context['rdf'] = self.as_rdf().serialize(format='json-ld', auto_compact=True,
+                                                 context=self.json_ld_context, indent=4)
+        context['breadcrumbs'] = self.get_breadcrumbs()
+        return context
+
+    def get_uri(self):
+        '''Get a URI for this page to use for making RDF assertions. Note that
+        this should return a full absolute path, e.g. with absolutize_url().'''
+        raise NotImplementedError
+
+    def as_rdf(self):
+        '''Generate an RDF graph representing the page.'''
+        # add the root node (this page)
+        graph = rdflib.Graph()
```

Comment:
I think this possibly should be a `ConjunctiveGraph` rather than just a `Graph`, but it seems to be working as is so may not matter. I saw something that made me think rdflib-jsonld might generate nicer output with a conjunctive graph, but when I tested I didn't see any difference. 

Also, I looked it up and you can bind a namespace like this:
`graph.bind('schema', SCHEMA_ORG)`

I think if you do that here, you don't have to specify context when you serialize. And it will make other serializations nicer too.

Reply:
I also tried using `ConjunctiveGraph` based on probably the same information you saw; it didn't make a different for me either. I'll try `graph.bind()`.

Reply:
update: `graph.bind()` didn't change anything, and didn't remove the need for a context. if I specify a context, the `schema:` prefixes are collapsed, but if I don't they aren't (and there are other problems too). 🤷‍♂